crates/oro-tree/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+use memmap::MmapOptions;
+use semver::{Version, VersionReq};
+use serde::{
+    de::{Deserializer, Error as SerdeError},
+    Deserialize,
+};
+use ssri::Integrity;
+use std::fs::File;
+use std::{collections::HashMap, path::Path};
+use thiserror::Error;
+
+fn default_as_false() -> bool {
+    false
+}
+
+fn parse_integrity<'de, D>(deserializer: D) -> Result<Integrity, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: &str = Deserialize::deserialize(deserializer)?;
+    s.parse().map_err(D::Error::custom)
+}
+
+fn parse_version<'de, D>(deserializer: D) -> Result<Version, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: &str = Deserialize::deserialize(deserializer)?;
+    s.parse().map_err(D::Error::custom)
+}
+
+fn parse_version_req<'de, D>(deserializer: D) -> Result<VersionReq, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: &str = Deserialize::deserialize(deserializer)?;
+    s.parse().map_err(D::Error::custom)
+}
+
+fn deserialize_requires<'de, D>(deserializer: D) -> Result<HashMap<String, VersionReq>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    struct Wrapper(#[serde(deserialize_with = "parse_version_req")] VersionReq);
+
+    let v = HashMap::<String, Wrapper>::deserialize(deserializer)?;
+    Ok(v.into_iter().map(|(k, Wrapper(v))| (k, v)).collect())
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Dependency {
+    #[serde(deserialize_with = "parse_version")]
+    version: Version,
+
+    #[serde(deserialize_with = "parse_integrity")]
+    integrity: Integrity,
+
+    #[serde(default = "default_as_false")]
+    dev: bool,
+
+    #[serde(default = "default_as_false")]
+    bundled: bool,
+
+    #[serde(default = "default_as_false")]
+    optional: bool,
```

Comment:
See comment above -- `#[serde(default)]` should be enough for these three fields.